src/main/java/com/wanderersoftherift/wotr/world/level/levelgen/processor/WeightedReplaceProcessor.java

```diff
@@ -0,0 +1,105 @@
+package com.wanderersoftherift.wotr.world.level.levelgen.processor;
+
+import com.wanderersoftherift.wotr.world.level.levelgen.processor.util.ProcessorUtil;
+import com.wanderersoftherift.wotr.world.level.levelgen.processor.util.StructureRandomType;
+import com.wanderersoftherift.wotr.world.level.levelgen.processor.util.WeightedBlockstateEntry;
+import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
+import com.mojang.serialization.codecs.RecordCodecBuilder;
+import net.minecraft.core.BlockPos;
+import net.minecraft.core.registries.BuiltInRegistries;
+import net.minecraft.tags.BlockTags;
+import net.minecraft.util.random.Weight;
+import net.minecraft.util.random.WeightedRandom;
+import net.minecraft.world.level.LevelReader;
+import net.minecraft.world.level.WorldGenLevel;
+import net.minecraft.world.level.block.Block;
+import net.minecraft.world.level.block.state.BlockState;
+import net.minecraft.world.level.levelgen.structure.templatesystem.StructurePlaceSettings;
+import net.minecraft.world.level.levelgen.structure.templatesystem.StructureProcessor;
+import net.minecraft.world.level.levelgen.structure.templatesystem.StructureProcessorType;
+import net.minecraft.world.level.levelgen.structure.templatesystem.StructureTemplate;
+
+import javax.annotation.Nullable;
+import java.util.List;
+
+import static com.wanderersoftherift.wotr.init.ModProcessors.WEIGHTED_REPLACE;
+import static com.wanderersoftherift.wotr.world.level.levelgen.processor.util.StructureRandomType.RANDOM_TYPE_CODEC;
+
+public class WeightedReplaceProcessor extends StructureProcessor {
+    public static final MapCodec<WeightedReplaceProcessor> CODEC = RecordCodecBuilder.mapCodec(builder ->
+            builder.group(
+                    WeightedBlockstateEntry.CODEC.listOf().fieldOf("gradient_list").forGetter(WeightedReplaceProcessor::getWeightList),
+                    BuiltInRegistries.BLOCK.byNameCodec().fieldOf("to_replace").forGetter(WeightedReplaceProcessor::getToReplace),
+                    RANDOM_TYPE_CODEC.optionalFieldOf("random_type", StructureRandomType.BLOCK).forGetter(WeightedReplaceProcessor::getStructureRandomType),
+                    Codec.LONG.optionalFieldOf("seed_adjustment", 6551687435L).forGetter(WeightedReplaceProcessor::getSeedAdjustment)
```

Comment:
Hark! A problem once gnawed at me, so the guidance of a witch I sought;
In the deep woods we met, by the light of flickering flame, and my question I wrought:
"What should be the default value of seed_adjustment?"
Into a trance she fell, and spoke as she shuddered and writhed:
6551687435!

I assume this is the way in which you chose this value :)